Document/0x05h-Testing-Platform-Interaction.md

```diff
@@ -149,23 +149,111 @@ Developers should take care to secure sensitive IPC components with the `signatu
 
 #### Overview
 
--- TODO [Provide a general description of the issue.] --
+Android apps can expose functionality to:
+* other apps via IPC mechanisms,
+* through custom URL schemes and
+* the user via the user interface.
+
+All input that is coming from these different sources cannot be trusted and need to be validated and/or sanitized. Validation ensures that only data is processed that the app is expecting. If validation is not enforced any input can be sent to the app, which might allow an attacker or malicious app to exploit vulnerable functionalities within the app.
 
 #### Static Analysis
 
--- TODO [Describe how to assess this given either the source code or installer package (APK/IPA/etc.), but without running the app. Tailor this to the general situation (e.g., in some situations, having the decompiled classes is just as good as having the original source, in others it might make a bigger difference). If required, include a subsection about how to test with or without the original sources.] --
+The source code should be checked if any functionality of the app is exposed, through:
+* Custom URL schemes: check also the test case "Testing Custom URL Schemes"
+* IPC Mechanisms: check also the test case "Testing Whether Sensitive Data Is Exposed via IPC Mechanisms"
```

Comment:
That's why I put the reference to the test case for IPC, that covers many of the IPC mechanisms.